src/dali/transaction_resolver.py

```diff
@@ -16,6 +16,7 @@
 from datetime import datetime
 from typing import Any, Dict, List, NamedTuple, Optional, cast
 
+from progressbar import ProgressBar, widgets
```

Comment:
Rather than importing the entire widget module, could you import the single classes that you need (Percentage, Bar, etc.)?